src/game_interpreter.cpp

```diff
@@ -1054,10 +1063,20 @@ bool Game_Interpreter::CommandControlVariables(RPG::EventCommand const& com) { /
 			// Characters
 			character = GetCharacter(com.parameters[5]);
 			if (character != NULL) {
+				int event_id = com.parameters[5];
 				switch (com.parameters[6]) {
 					case 0:
 						// Map ID
-						value = character->GetMapId();
+						if (!Player::IsRPG2k()
+								|| event_id == Game_Character::CharPlayer
+								|| event_id == Game_Character::CharBoat
+								|| event_id == Game_Character::CharShip
+								|| event_id == Game_Character::CharAirship) {
+							value = character->GetMapId();
+						} else {
+							// This is an RPG_RT bug for 2k only. Requesting the map id of an event always returns 0.
+							value = 0;
```

Comment:
This will "break" when you load a 2k3 save in 2k until you load a new map. But not worth to support this.